### PR TITLE
Remove Spree::StockLocation

### PR DESCRIPTION
**Pull request by mbj**
This model is totally anemic meanwhile.

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb This is ready for review.

**Comment by dkubb** [comment]
@mbj I found a few more places that *may* be related to stock location:

```diff
diff --git a/backend/app/assets/stylesheets/spree/backend/sections/_orders.scss b/backend/app/assets/stylesheets/spree/backend/sections/_orders.scss
index d4bd838..54fb07c 100644
--- a/backend/app/assets/stylesheets/spree/backend/sections/_orders.scss
+++ b/backend/app/assets/stylesheets/spree/backend/sections/_orders.scss
@@ -20,16 +20,6 @@
   }
 }
 
-[data-hook="admin_order_form_fields"] {
-  legend.stock-location {
-    color: $color-body-text;
-
-    .stock-location-name {
-      color: $color-success;
-    }
-  }
-}
-
 // Customize orduct add fieldset
 #add-line-item {
   fieldset {
diff --git a/backend/app/views/spree/admin/shared/_translations.html.erb b/backend/app/views/spree/admin/shared/_translations.html.erb
index c99e1a7..b67f9a4 100644
--- a/backend/app/views/spree/admin/shared/_translations.html.erb
+++ b/backend/app/views/spree/admin/shared/_translations.html.erb
@@ -17,7 +17,6 @@
     :first_day                => Spree.t(:first_day,
                                          :scope => 'date_picker',
                                          :default => 0).to_i,
-    :item_stock_placeholder   => Spree.t(:choose_location),
     :loading                  => Spree.t(:loading),
     :month_names              => I18n.t(:month_names, :scope => :date).reject(&:blank?),
     :more                     => Spree.t(:more),
diff --git a/core/config/locales/en.yml b/core/config/locales/en.yml
index bd5798f..a7e52a5 100644
--- a/core/config/locales/en.yml
+++ b/core/config/locales/en.yml
@@ -380,7 +380,6 @@ en:
     choose_a_customer: Choose a customer
     choose_currency: Choose Currency
     choose_dashboard_locale: Choose Dashboard Locale
-    choose_location: Choose location
     city: City
     click_and_drag_on_the_products_to_sort_them: Click and drag on the products to sort them.
     clone: Clone
@@ -589,7 +588,6 @@ en:
     listing_users: Listing Users
     loading: Loading
     locale_changed: Locale Changed
-    location: Location
     lock: Lock
     logs: "Logs"
     logged_in_as: Logged in as
@@ -615,7 +613,6 @@ en:
     minimal_amount: Minimal Amount
     month: Month
     more: More
-    move_stock_between_locations: Move Stock Between Locations
     my_account: My Account
     my_orders: My Orders
     name: Name
``

**Comment by mbj** [comment]
good catch, fixing these also.

**Comment by dkubb** [comment]
@mbj I noticed the first parameter in the `Spree::Package` constructor is `stock_location`. Should this be changed to `address` to avoid confusion?

**Comment by mbj** [comment]
> @mbj I noticed the first parameter in the Spree::Package constructor is stock_location. Should this be changed to address to avoid confusion?

No as `spree_active_shipping` will accept `Package#stock_location` but than use `build_location` that can handle both `Spree::Address` and `Spree::StockLocation` objects.

As we are about to abandon `spree_active_shipping` I did not touch it to remove that artifact.

File: frontend/spec/controllers/spree/checkout_controller_spec.rb [hunk 1]
```diff
@@ -585,9 +585,7 @@ def do_request
 
 # Legacy DB infected spec not suitable yet for mutation testing and with weak coverage
 describe Spree::CheckoutController, type: :controller, db: :transaction, mutant: false do
-  let!(:store)          { create(:store, default: true) }
-  let!(:stock_location) { create(:stock_location)       }
-
+  let!(:store)      { create(:store, default: true) }
```
Comment (dkubb) [inline frontend/spec/controllers/spree/checkout_controller_spec.rb]: The `}` should be aligned with the other lines in this block.

**Comment by mbj** [comment]
@dkubb Ready for the next pass.

**Comment by dkubb** [comment]
@mbj LGTM. I will merge when green.

**Pull request merged**